sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -249,16 +249,19 @@ def delete_expired_environments(self) -> t.List[Environment]:
 
     def delete_snapshots(self, snapshot_ids: t.Iterable[SnapshotIdLike]) -> None:
         self.engine_adapter.delete_from(
-            self.snapshots_table, where=self._snapshot_id_filter(snapshot_ids)
+            self.snapshots_table, where=self._snapshot_id_filter(snapshot_ids, self.snapshots_table)
         )
 
     def snapshots_exist(self, snapshot_ids: t.Iterable[SnapshotIdLike]) -> t.Set[SnapshotId]:
         return {
             SnapshotId(name=name, identifier=identifier)
             for name, identifier in self.engine_adapter.fetchall(
-                exp.select("name", "identifier")
+                exp.select(
+                    exp.to_column(f"{self.snapshots_table}.name"),
```

Comment:
Did you add this because MSSQL requires the full qualified name?

Reply:
Annoyingly, the joins using `ON` with qualified names means we have to qualify them everywhere